lib/ash/data_layer/ets/ets.ex

```diff
@@ -742,23 +742,37 @@ defmodule Ash.DataLayer.Ets do
         end)
 
       :avg ->
-        records
-        |> then(fn records ->
-          if uniq? do
-            records
-            |> Stream.map(&Map.get(&1, field))
-            |> Stream.uniq()
+        items =
+          records
+          |> then(fn records ->
+            if uniq? do
+              records
+              |> Stream.map(&Map.get(&1, field))
+              |> Stream.uniq()
+            else
+              records
+              |> Stream.map(&Map.get(&1, field))
+            end
+          end)
+
+        first_item = List.first(Enum.to_list(Stream.take(items, 1)))
+
+        initial_value =
+          if is_struct(first_item, Decimal) do
+            Decimal.new(0)
           else
-            records
-            |> Stream.map(&Map.get(&1, field))
-            |> Stream.uniq()
```

Comment:
This is unrelated to `decimal`, but I removed the `Stream.uniq` here when `uniq?` is false.